### PR TITLE
should check and return err when err of r is not empty

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

should check and return err when err of r is not empty

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by wackxu** [comment]
kindly ping @deads2k  @derekwaynecarr for approval

File: pkg/kubectl/cmd/certificates.go [hunk 1]
```diff
@@ -175,6 +175,10 @@ func (options *CertificateOptions) modifyCertificateCondition(f cmdutil.Factory,
 		Flatten().
 		Latest().
 		Do()
+	err = r.Err()
```
Comment (derekwaynecarr) [inline pkg/kubectl/cmd/certificates.go]: i see a mixed pattern on how we chose to handle this, but isnt the visit function effectively ending up doing the same thing internally?

Comment (wackxu) [reply pkg/kubectl/cmd/certificates.go]: @derekwaynecarr yeah, in the visit function there also has err check. But like other command, if the `Result` has err, we should end up immediately.

**Pull request closed without merging**